source/Calamari.Shared/Integration/Processes/CommandLineRunner.cs

```diff
@@ -50,6 +56,22 @@ public CommandResult Execute(CommandLineInvocation invocation)
             }
         }
 
+        protected virtual List<ICommandOutput> GetCommandOutputs(CommandLineInvocation invocation)
```

Comment:
Instead of the caller creating the same two command outputs each time (console and servicemessage), it's controlled through `CommandLineInvocation`. This means `ConsoleCommandOutput` and `ServiceMessageCommandOutput` don't have to be shared.